lighting/specular.hlsl

```diff
@@ -28,5 +27,5 @@ license:
 
 #ifndef FNC_SPECULAR
 #define FNC_SPECULAR
-float specular(ShadingData shadingData) { return SPECULAR_FNC(shadingData); }
+float3 specular(ShadingData shadingData) { return float3(1.0, 1.0, 1.0) * SPECULAR_FNC(shadingData); }
```

Comment:
curious about this. Why returning a vector over a float?

Reply:
Great question. The specular highlight of metals is coloured, unlike that of dielectrics which is monochromatic. This is why the f0 of dielectrics is a scalar value (typically 0.04) while the f0 of metals in our workflow is the colour of the specular (which we input as the albedo property).

It then comes as no surprise that the fresnel component of the Cook-Torrance BRDDF returns a vector, previously this was handled with this horrendous hack `float F = fresnel(vec3(_fresnel), LoH).r;` which took the red channel of the coloured specular of turned it into a monochromatic value.
We're now correctly returning the specular highlight as an RGB value.